corehq/apps/app_manager/app_strings.py

```diff
@@ -166,6 +166,26 @@ def _create_module_details_app_strings(module, langs):
                         ),
                         clean_trans(item.value, langs)
                     )
+            elif column.format in ('enum-image', 'clickable-icon'):
+                for item in column.enum:
+                    yield (
+                        id_strings.detail_column_enum_variable(
+                            module,
+                            detail_type,
+                            column,
+                            item.key_as_variable,
+                        ),
+                        clean_trans(item.value, langs)
+                    )
+                    yield (
```

Comment:
If you added this `yield` in the original `for item in column.enum` behind an if statement you would not have to duplicate the code for the first one. Or extract it into a function.

Reply:
Good call, thanks https://github.com/dimagi/commcare-hq/pull/34044/commits/918093e3e64e9e9e4c15c2a6461f10a24ff47006